README.md

```diff
@@ -28,16 +31,151 @@ brew install please
+deploying code to a local cluster, and pushing it to a remote registry. 
+
+The [genrule()](https://please.build/codelabs/genrule) codelab covers 
+extending Please with custom build definitions to truly automate any 
+part of your deployment process. 
+
+Why Please, and not make?
+===============
 
+Make is a great tool for running tasks. It's easy enough to understand
+because it leaves you very close to the shell. The problem is, it has 
+limited capability to build out complexity. There have been attempts to 
+generate make files from higher level tools like cmake and ninja but they
+fall short of what Please sets out to achieve. 
+
+The Please build language is a full programming language. There are a high 
+level set of build rules that make up a declarative DSL to define build 
+targets, however you can drop into an imperative language that resembles 
+python when necessary:
+
+```python
+subinclude("//my_custom_defs:markdown_page")
+
+pages = []
+for page in glob(include = ["*.md"]):
+    pages += markdown_page(
+        name = trimsuffix(page),
+        srcs = [page],
+        visibility = ["//website/..."],
+    )   
+
+go_binary (
+    name = "webserver",
+    srcs = ["main.go"],
+    deps = ["//third_party/go:protobuf"],
+    data = pages,
+    visibility = ["//services/foo/..."],
+)
+```
+
+This is distinctively operating at a higher level when compared to make:
+
+```
+protobuf:
+    go install google.golang.org/protobuf
+
+webserver: protobuf
+    go tool compile --pack foo.go -o foo.a
+
+pages: ???
+```
+
+Additionally, `make` builds are not hermetic. The above make example installs
+protobuf into the host machines Go path. Please builds only have access to files 
+and environment variables they have explicitly been given access to. You can play 
+around in the environment targets are built in with `plz build //some/target --shell`. 
+Additionally, on linux systems, Please can take this a step further with Linux 
+[namespaces](https://en.wikipedia.org/wiki/Linux_namespaces) to improve 
+sandboxing especially of the network. Please also has built in task 
+parallelism so can take full advantage of multi-core machines which were 
+not a consideration 40 years ago when make was designed. 
+
+Finally, Please has a robust caching mechanism base on hashing the 
+inputs of each rule. Makes cache invalidation is based on the last
+modified timestamp which can change unexpectedly forwards and 
+backwards in time. Combine this with hermetic builds, and Please 
+caching almost never incorrect.  
+ 
+
+Why Please, not Bazel, Buck or Pants? 
+=============== 
+
+These build systems are all very similar so choosing between them can be hard. 
+Please was built to replace Buck internally here at Thought Machine. We found Buck 
+didn't deliver what Blaze did at google and was very hard to extend to other 
+languages not supported as first class by Buck itself. 
+
+Please was originally transparently replaced Buck. It was feature compatible 
```

Comment:
"was originally a replacement for Buck to be feature compatible with the subset of features we used"